docs/proposals/2628-kai-scheduler/README.md

```diff
@@ -0,0 +1,89 @@
+  - [Integration with KAI PodGrouper](#integration-with-kai-podgrouper)
+  - [Queue Management and Labeling](#queue-management-and-labeling)
+  - [Validation and Error Handling](#validation-and-error-handling)
+
+## Summary
+
+This Kubeflow Enhancement Proposal (KEP) aims to integrate [KAI Scheduler](https://github.com/NVIDIA/KAI-Scheduler), NVIDIA's open-source scheduler designed specifically for GPU workloads, into Kubeflow Trainer V2. The integration will extend the existing PodGroupPolicy API to support KAI as a gang-scheduling option alongside the currently supported Volcano and Co-Scheduling schedulers. This enhancement will enable efficient resource allocation and scheduling for distributed training workloads using TrainJob resources in Kubeflow environments.
+
+## Motivation
+
+[KAI Scheduler](https://github.com/NVIDIA/KAI-Scheduler) has emerged as a specialized scheduling solution for GPU workloads, offering features specifically designed to optimize the scheduling of machine learning training jobs. Currently, KAI only integrates with Kubeflow Training Operator v1, limiting its accessibility to users who have migrated to or prefer using Kubeflow Trainer V2. The lack of KAI support in Trainer V2 creates a gap in scheduling options for users who require advanced scheduling capabilities.
+
+The existing PodGroupPolicy API in Trainer V2 already supports multiple schedulers, demonstrating the framework's extensibility. Adding KAI support would provide users with another powerful scheduling option, particularly beneficial for environments where NVIDIA hardware is prevalent and where specialized GPU workload scheduling can significantly improve resource utilization and training performance.
+
+### Goals
+
+- Extend the PodGroupPolicy API in Trainer V2 to support KAI Scheduler as a gang-scheduling option
+- Implement a KAI plugin that integrates with the existing framework architecture
+- Enable automatic creation of scheduling queues and proper labeling for KAI-scheduled workloads
+- Provide comprehensive documentation and examples for using KAI with TrainJob resources
+- Ensure compatibility with JobSet orchestration used by Trainer V2
+- Maintain consistency with existing scheduler integrations (Co-Scheduling)
+
+
+## Design Details
+
+The implementation will follow the established pattern used for other scheduler integrations in Trainer V2, extending the PodGroupPolicySource struct and creating a dedicated plugin for KAI Scheduler functionality.
+
+### API Extensions
+
+The PodGroupPolicySource struct in `trainingruntime_types.go` will be extended to include KAI Scheduler configuration. This follows the same pattern as existing scheduler integrations:
+
+```go
+type PodGroupPolicySource struct {
+    CoScheduling *CoSchedulingConfig `json:"coscheduling,omitempty"`
+    KAI          *KAIConfig          `json:"kai,omitempty"`
+}
+
+type KAIConfig struct {
+    Queue string `json:"queue,omitempty"`
```

Comment:
This is going to be really confusing to someone who uses KAI + Kueue.

cc @romanbaron.

Is there anyway to use gang scheduling in KAI without needing a KAI queue?

Volcano is used for PodGroup functionality in Trainer but I don't think one has to use Volcano Queues to use this.

Reply:
Each KAI PodGroup must reference a designated KAI scheduling queue to clarify which queue’s resources are allocated to it.

One approach to integrating KAI with Kueue is to configure a single, unlimited queue for KAI, allowing Kueue to govern resources and dispatch workloads. Under this model, KAI schedules only the workloads that have been dispatched (i.e., pods are created and ungated), while fairness enforcement remains outside its scope.

@kannon92 - does it sounds like a possible solution? If not, we can discuss the possible options further.